mrc/localization/camera/utils/connected_components_detector.py

```diff
@@ -0,0 +1,75 @@
+import math
+import cv2
+from mrc.utils.maths import points_2d_sqr_distance, point_in_rectangle, rescale_rectangle
+
+
+class ConnectedComponentsDetector:
+    def __init__(self):
+        pass
+
+    def detect(self, image, remove_background=True, remove_irrelevant=True):
+        num_labels, labels, stats, centroids = cv2.connectedComponentsWithStats(image)
+        stats_list = stats.tolist()
+        centroids_list = centroids.tolist()
+        if remove_background:
+            background_component_index = self._get_background_component_index(image, stats, centroids)
+            stats_list.pop(background_component_index)
+            centroids_list.pop(background_component_index)
+        if remove_irrelevant and remove_background:
+            irrelevant_indexes = self._get_irrelevant_components_indexes(stats_list, centroids_list, 500, 25)
+            stats_list = [stats_list[i] for i in range(len(stats_list)) if i not in irrelevant_indexes]
+            centroids_list = [centroids_list[i] for i in range(len(centroids_list)) if i not in irrelevant_indexes]
+        return stats_list, centroids_list
+
+    def _get_background_component_index(self, image, stats, centroids):
+        image_centre = (image.shape[0] / 2, image.shape[1] / 2)
+        areas = [s[cv2.CC_STAT_HEIGHT] * s[cv2.CC_STAT_WIDTH] for s in stats]
```

Comment:
- [ ] Why not`: areas = [x *y for x,y,_,_,_ in stats]`

Reply:
What if it changes with OpenCV version?